primitives/inherents/src/lib.rs

```diff
@@ -410,13 +410,19 @@ pub trait ProvideInherent {
 	/// Create an inherent out of the given `InherentData`.
 	fn create_inherent(data: &InherentData) -> Option<Self::Call>;
 
-	/// If `Some`, indicates that an inherent is required. Check will return the inner error if no
-	/// inherent is found. If `Err`, indicates that the check failed and further operations should
-	/// be aborted.
+	/// Determines whether this inherent is required in this block.
+	/// Ok(None) indicates that this inherent is not required in this block. The default implementation returns this.
+	/// Ok(Some(e)) indicates that this inherent is required in this block, and a call to `check_inherent`
```

Comment:
I don't think this is quite right, but I couldn't find a better understanding. What actually calls this function and uses the inner error value?

Also, why does the timestamp pallet not use this `is_inherent_required` function?

Reply:
impl_outer_inherent, will make use of it in its implementation of check_extrinsics.

Note that is_inherent_required doesn't check if there has been multiple inherent provided instead of one.
Also it has a broken behavior when mixed with validate unsigned transactions. See https://github.com/paritytech/substrate/issues/6243